framework/operator_test.go

```diff
@@ -0,0 +1,364 @@
+package framework
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cenk/backoff"
+)
+
+type testResource struct {
+	Error       error
+	ErrorCount  int
+	ErrorMethod string
+	Order       []string
+
+	errorCount int
+}
+
+func (r *testResource) GetCurrentState(obj interface{}) (interface{}, error) {
+	m := "GetCurrentState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return nil, r.Error
+	}
+
+	return nil, nil
+}
+
+func (r *testResource) GetDesiredState(obj interface{}) (interface{}, error) {
+	m := "GetDesiredState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return nil, r.Error
+	}
+
+	return nil, nil
+}
+
+func (r *testResource) GetCreateState(obj, currentState, desiredState interface{}) (interface{}, error) {
+	m := "GetCreateState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return nil, r.Error
+	}
+
+	return nil, nil
+}
+
+func (r *testResource) GetDeleteState(obj, currentState, desiredState interface{}) (interface{}, error) {
+	m := "GetDeleteState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return nil, r.Error
+	}
+
+	return nil, nil
+}
+
+func (r *testResource) ProcessCreateState(obj, createState interface{}) error {
+	m := "ProcessCreateState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return r.Error
+	}
+
+	return nil
+}
+
+func (r *testResource) ProcessDeleteState(obj, deleteState interface{}) error {
+	m := "ProcessDeleteState"
+	r.Order = append(r.Order, m)
+
+	if r.returnErrorFor(m) {
+		return r.Error
+	}
+
+	return nil
+}
+
+func (r *testResource) returnErrorFor(errorMethod string) bool {
+	ok := r.Error != nil && r.ErrorCount > r.errorCount && r.ErrorMethod == errorMethod
+
+	if ok {
+		r.errorCount++
+		return true
+	}
+
+	return false
+}
+
+// Test_Operator_ProcessCreate_NoResource ensures there is an error thrown when
+// executing ProcessCreate without having any resources provided.
+func Test_Operator_ProcessCreate_NoResource(t *testing.T) {
+	err := testMustNewFramework(t).ProcessCreate(nil, nil)
+	if !IsExecutionFailed(err) {
+		t.Fatal("expected", true, "got", false)
+	}
+}
```

Comment:
Here the diff is bad again. Half of the tests are unchanged and have been there already. I only added more tests towards the retry logic below.